pingora-core/src/protocols/http/compression/mod.rs

```diff
@@ -571,14 +593,94 @@ fn compressible(resp: &ResponseHeader) -> bool {
+                // This is equivalent to slice.trim_ascii() which is unstable
+                while let [first, rest @ ..] = v {
+                    if first.is_ascii_whitespace() {
+                        v = rest;
+                    } else {
+                        break;
+                    }
+                }
+                while let [rest @ .., last] = v {
+                    if last.is_ascii_whitespace() {
+                        v = rest;
+                    } else {
+                        break;
+                    }
+                }
+                v
+            })
+            .any(|v| v == b"*" || v.eq_ignore_ascii_case(value.as_ref()));
+
+        // Prepend our value to the existing header unless already present
+        if !already_present {
+            let mut header = Vec::new();
+            header.extend_from_slice(value.as_ref());
+            header.extend_from_slice(b", ");
+            header.extend_from_slice(existing.as_bytes());
+            resp.insert_header(&VARY, HeaderValue::from_bytes(&header).unwrap())
+                .unwrap();
+        }
+    } else {
+        resp.insert_header(&VARY, HeaderValue::from_name(value.clone()))
+            .unwrap();
+    }
+}
+
+#[test]
+fn test_add_vary_header() {
+    let mut header = ResponseHeader::build(200, None).unwrap();
+    add_vary_header(&mut header, &http::header::ACCEPT_ENCODING);
+    assert_eq!(header.headers.get("Vary").unwrap(), "accept-encoding");
+
+    let mut header = ResponseHeader::build(200, None).unwrap();
+    header.insert_header("Vary", "Accept-Language").unwrap();
+    add_vary_header(&mut header, &http::header::ACCEPT_ENCODING);
+    assert_eq!(
+        header.headers.get("Vary").unwrap(),
+        "accept-encoding, Accept-Language"
+    );
+
+    let mut header = ResponseHeader::build(200, None).unwrap();
+    header
+        .insert_header("Vary", "Accept-Language, Accept-Encoding")
+        .unwrap();
+    add_vary_header(&mut header, &http::header::ACCEPT_ENCODING);
+    assert_eq!(
+        header.headers.get("Vary").unwrap(),
+        "Accept-Language, Accept-Encoding"
+    );
+
+    let mut header = ResponseHeader::build(200, None).unwrap();
+    header.insert_header("Vary", "*").unwrap();
+    add_vary_header(&mut header, &http::header::ACCEPT_ENCODING);
+    assert_eq!(header.headers.get("Vary").unwrap(), "*");
+}
+
 fn adjust_response_header(resp: &mut ResponseHeader, action: &Action) {
-    use http::header::{HeaderValue, CONTENT_ENCODING, CONTENT_LENGTH, TRANSFER_ENCODING};
+    use http::header::{
+        HeaderValue, ACCEPT_RANGES, CONTENT_ENCODING, CONTENT_LENGTH, TRANSFER_ENCODING,
+    };
 
     fn set_stream_headers(resp: &mut ResponseHeader) {
         // because the transcoding is streamed, content length is not known ahead
         resp.remove_header(&CONTENT_LENGTH);
-        // remove Accept-Ranges header because range requests will no longer work
-        resp.remove_header(&ACCEPT_RANGES);
```

Comment:
We'd prefer the removal of this header rather than inserting the `none`.

Reply:
Oh well… Reverted that change.